addons/mail/static/src/services/messaging_service/messaging_service.js

```diff
@@ -0,0 +1,127 @@
+/** @odoo-module **/
+
+import ModelManager from '@mail/model/model_manager';
+
+import { registry } from '@web/core/registry';
+
+const { Store } = owl;
+const { EventBus } = owl.core;
+
+class MessagingService {
+
+    constructor(env) {
+        this.env = env;
+        /**
+         * Messaging store
+         */
+        const store = new Store({
+            env,
+            state: {
+                messagingRevNumber: 0,
+            },
+        });
+        /**
+         * Registry of models.
+         */
+        this.models = {};
+        /**
+         * Environment keys used in messaging.
+         */
+        Object.assign(this, {
+            autofetchPartnerImStatus: true,
+            destroyMessaging() {
+                if (this.modelManager) {
+                    this.modelManager.deleteAll();
+                    this.messaging = undefined;
+                }
+            },
+            disableAnimation: false,
+            isMessagingInitialized() {
+                if (!this.messaging) {
+                    return false;
+                }
+                return this.messaging.isInitialized;
+            },
+            /**
+             * States whether the environment is in QUnit test or not.
+             *
+             * Useful to prevent some behaviour in QUnit tests, like applying
+             * style of attachment that uses url.
+             */
+            isQUnitTest: false,
+            loadingBaseDelayDuration: 400,
+            messaging: undefined,
+            messagingBus: new EventBus(),
+            /**
+             * Promise which becomes resolved when messaging is created.
+             *
+             * Useful for discuss widget to know when messaging is created, because this
+             * is an essential condition to make it work.
+             */
+            messagingCreatedPromise: this.createMessaging(),
```

Comment:
Is it still necessary? Possible to use the service booting code to wait on this without a custom promise? Maybe with a new messaging_boot service that would be different than messaging itself?